libs/MeshKernel/include/MeshKernel/Mesh2D.hpp

```diff
@@ -236,6 +227,12 @@ namespace meshkernel
         ///  @brief Classifies the nodes (makenetnodescoding)
         void ClassifyNodes();
 
+        /// @brief Get the node type
+        MeshNodeType GetNodeType(const UInt nodeId) const { return m_nodesTypes[nodeId]; }
+
+        /// @brief Get the node type
+        void GetNodeTypes(std::vector<MeshNodeType>& nodeTypes) const { nodeTypes = m_nodesTypes; }
```

Comment:
i think this is equivalent of returning a  reference (const or not depend on the use case)

const std::vector<MeshNodeType>& GetNodeTypes() const { return m_nodesTypes; }

I find this better because you do need one line less at the caller side